src/Cli/Microsoft.TemplateEngine.Cli/Commands/install/InstallCommandArgs.cs

```diff
@@ -2,38 +2,50 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.CommandLine;
+using NuGet.Packaging.Core;
 
 namespace Microsoft.TemplateEngine.Cli.Commands
 {
+    internal interface ITemplateIdentifierArgument;
+
+#pragma warning disable CS9113 // Parameter is unread.
+    internal record FileBasedTemplateIdentifier(FileInfo File) : ITemplateIdentifierArgument;
+#pragma warning restore CS9113 // Parameter is unread.
+
+#pragma warning disable CS9113 // Parameter is unread.
+    internal record PackageIdentifier(PackageIdentity Package) : ITemplateIdentifierArgument;
+#pragma warning restore CS9113 // Parameter is unread.
+
```

Comment:
This is the core logical change - introducing structures to represent the allowed variants of input for package identifiers.
With these defined, we have to then change _parsing_ to create these structures, and change _execution_ to handle the structures potentially-differently.